src/components/TodoItem.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { deleteTodo, updateTodo } from '../api/todos';
+import { EventType } from '../types/types';
+
+interface Props {
+  todo: Todo,
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+  setErrorMessage: React.Dispatch<React.SetStateAction<string | null>>,
+  todosLoader: boolean,
+  isLoadingCompleted: boolean,
+}
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  setTodos,
+  setErrorMessage,
+  todosLoader,
+  isLoadingCompleted,
+}) => {
+  const { completed, title, id } = todo;
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, []);
+  const handleCheckboxClick = async () => {
+    setIsLoading(true);
+
+    if (completed) {
+      await updateTodo(
+        todo.id,
+        { completed: false },
+        setTodos,
+        setErrorMessage,
+      );
+    } else {
+      await updateTodo(
+        todo.id,
+        { completed: true },
+        setTodos,
+        setErrorMessage,
+      );
+    }
```

Comment:
you can simply assign value of completed to object with key completed, or use ternary at least

Reply:
yup. didn't get it